src/libraries/System.Text.Json/src/System/Text/Json/Serialization/ReflectionEmitMemberAccessor.cs

```diff
@@ -65,7 +65,7 @@ internal sealed class ReflectionEmitMemberAccessor : MemberAccessor
 
             Debug.Assert(type != null);
             Debug.Assert(!type.IsAbstract);
-            Debug.Assert(type.GetConstructors(BindingFlags.Public | BindingFlags.Instance).Contains(constructor));
+            Debug.Assert(HasConstructor(type, constructor));
```

Comment:
I would expect this to fail to compile in a release build.

Reply:
Yep. Thanks